core/src/main/java/hudson/Util.java

```diff
@@ -1927,4 +1927,23 @@ public static long daysElapsedSince(@NonNull Date date) {
     private static PathRemover newPathRemover(@NonNull PathRemover.PathChecker pathChecker) {
         return PathRemover.newFilteredRobustRemover(pathChecker, DELETION_RETRIES, GC_AFTER_FAILED_DELETE, WAIT_BETWEEN_DELETION_RETRIES);
     }
+
+    /**
+     * Returns SHA-256 Digest of input bytes
+     */
+    @Restricted(NoExternalUse.class)
+    public static byte[] getSHA256DigestOf(@NonNull byte[] input)
+    {
+        try {
+            if (input != null) {
+                MessageDigest sha256Digest = MessageDigest.getInstance("SHA-256");
+                sha256Digest.update(input);
+                return sha256Digest.digest();
+            }
+        } catch (NoSuchAlgorithmException noSuchAlgorithmException) {
+            LOGGER.log(Level.WARNING, "Failed to instantiate SHA-256 message digest algorithm", noSuchAlgorithmException);
+            throw new IllegalStateException("Failed to instantiate SHA-256 message digest algorithm");
```

Comment:
there should not be a need to both log and throw (especially if no extra information is being added to the logs) - if you are throwing this then where it is caught should eventually log it.

Additionally don't loose the original exception.

```suggestion
            throw new IllegalStateException("SHA-256 could not be instantiated, but is required to be implemented by the language specification", noSuchAlgorithmException);
```